util/conditions/v1beta2/sort.go

```diff
@@ -38,26 +46,90 @@ func defaultSortLessFunc(i, j metav1.Condition) bool {
 		return fi < fj
 	}
 
-	li, oki := last[i.Type]
-	lj, okj := last[j.Type]
-	switch {
-	case oki && !okj:
-		return false
-	case !oki && okj:
-		return true
-	case oki && okj:
-		return li < lj
-	}
-
 	return i.Type < j.Type
 }
 
-var first = map[string]int{
-	clusterv1.AvailableV1Beta2Condition: 0,
-	clusterv1.ReadyV1Beta2Condition:     1,
+// The order array below leads to the following condition ordering:
+//
+// | Condition                      | Cluster | KCP | MD | MS | MP | Machine |
+// |--------------------------------|---------|-----|:---|----|----|---------|
+// | -- Availability conditions --  |         |     |    |    |    |         |
+// | Available                      | x       | x   | x  |    | x  | x       |
+// | Ready                          |         |     |    |    |    | x       |
+// | RemoteConnectionProbe          | x       |     |    |    |    |         |
+// | BootstrapConfigReady           |         |     |    |    | x  | x       |
+// | InfrastructureReady            | x       |     |    |    | x  | x       |
+// | ControlPlaneInitialized        | x       |     |    |    |    |         |
+// | ControlPlaneAvailable          | x       |     |    |    |    |         |
+// | WorkersAvailable               | x       |     |    |    |    |         |
+// | CertificatesAvailable          |         | x   |    |    |    |         |
+// | Initialized                    |         | x   |    |    |    |         |
+// | EtcdClusterHealthy             |         | x   |    |    |    |         |
+// | ControlPlaneComponentsHealthy  |         | x   |    |    |    |         |
+// | NodeHealthy                    |         |     |    |    |    | x       |
+// | NodeReady                      |         |     |    |    |    | x       |
+// | APIServerPodHealthy            |         |     |    |    |    | x       |
+// | ControllerManagerPodHealthy    |         |     |    |    |    | x       |
+// | SchedulerPodHealthy            |         |     |    |    |    | x       |
+// | EtcdPodHealthy                 |         |     |    |    |    | x       |
+// | EtcdMemberHealthy              |         |     |    |    |    | x       |
```

Comment:
What about moving Etcd conditions right before API server (if etcd is broken, it should be fixed before the API server)